sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_options.hpp

```diff
@@ -104,6 +104,12 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
        * @brief Holds the customer provided key used when making requests.
        */
       Azure::Nullable<EncryptionKey> CustomerProvidedKey;
+
+      /**
+       * The filesystem url. This is only non-null for directory clients that are created from a
+       * filesystem client, so that this directory client knows where to send ListPaths requests.
+       */
+      Azure::Nullable<Azure::Core::Url> FileSystemUrl;
```

Comment:
We have previously recommended to use `std::string` for urls, instead of `Core::Url`, for ease-of-use based on interoperability.

That is why all of our public url fields/parameters accept string (or `Nullable<string>` if necessary). We could consider revisiting this guidance, but that's the strategy so far.

Reply:
cc @JeffreyRichter if you have additional thoughts on this (or new insights)

Reply:
Does it matter that this is a type in the `_detail` namespace, and thus NOT a part of the public API contract?

Reply:
Didn't notice the namespace. In that case, using `Core::Url` is reasonable :)

Reply:
I had a similar concern about this field (and left a now deleted comment) until I noticed that :).

Reply:
The change in this PR is about internal implementation and doesn't affect public interface.